rav1e_js/www/src/App.tsx

```diff
@@ -7,88 +7,48 @@
-const enc = new Encoder(conf);
-
-// Frame to send to encoder
-const f = enc.newFrame();
+import { ChromaSampling, EncoderConfig, Frame, VideoEncoder } from "rav1e";
 
 export default function App() {
-	const [encoderStatus, setEncoderStatus] = useState(EncoderStatus.Initialized);
-	const [flushing, setFlushing] = useState(false);
-	const [frameQ_length, setFrameQ_length] = useState(0);
-	const [packets, setPackets] = useState(Array<Packet>())
-
-	const sendFrame = () => {
-		enc.sendFrame(f);
-		setFrameQ_length(frameQ_length + 1);
-		setEncoderStatus(EncoderStatus.Received)
-	};
-
-	const flush = () => {
-		enc.flush();
-		setFlushing(true);
-		setEncoderStatus(EncoderStatus.Flushed);
-	}
-
-	const receivePacket = () => {
-		try {
-			const p = enc.receivePacket();
-			setEncoderStatus(EncoderStatus.Encoded);
-
-			// append new packet
-			const newPackets = Array.from(packets);
-			newPackets.push(p);
-			setPackets(newPackets);
-
-			// needed to receive the "limit reached" status
-			if (flushing && (frameQ_length - newPackets.length) === 0) {
-				receivePacket();
-			}
-		} catch (e) {
-			if (e === "encoded") {
-				receivePacket();
-			} else if (e === "need more data") {
-				setEncoderStatus(EncoderStatus.NeedMoreData);
-			} else if (e === "limit reached") {
-				setEncoderStatus(EncoderStatus.LimitReached);
-			} else {
-				setEncoderStatus(e);
-				console.error(e);
+	// useEffect, because then it runs after the browser rendered the display content
+	useEffect(() => {
+		const video = document.getElementById("video") as HTMLVideoElement;
+		console.log(video.videoWidth, video.videoHeight);
+
+		video.onended = (e) => {
+			enc.flush();
+			console.log("flushed")
+
+			// encode all frames
+			while (true) {
+				try {
+					const p = enc.receivePacket();
+					console.log(p.display());
+				} catch (e) {
+					if (e === "encoded") {
+						console.warn(e);
+					} else if (e === "limit reached") {
+						console.warn(e);
+						break;
+					} else {
+						console.error(e);
+					}
+				}
 			}
-		}
-	}
-
-	return (
-		<div>
-			<h1>Rav1e_js demo</h1>
-
-			<button disabled={flushing} onClick={sendFrame}>Send Frame</button>
-			<button disabled={flushing || packets.length === frameQ_length} onClick={flush}>Flush</button>
-			<button disabled={packets.length === frameQ_length} onClick={receivePacket}>Receive Packet</button>
-
-			<p>Frames in queue: {frameQ_length - packets.length}</p>
-			<p>Encoder Status: {encoderStatus}</p>
-
-			<HelpAndAbout />
-			<EncoderConfigDetails configStr={configStr} />
-
-			{packets.length !== 0 ? <h3>Encoded packets</h3> : <div></div>}
-			<ol start={0} >{packets.map((p) => <li key={p.display()}>{p.display()}</li>)}</ol>
-		</div>
-	);
+		};
+
+		// configure encoder
+		const enc = new VideoEncoder(
+			new EncoderConfig()
+				.setSpeed(10)
+				.setDim(video.videoWidth, video.videoHeight)
+				// .setColorDescription(...) (is not available yet)
+				.setChromaSampling(ChromaSampling.Cs444)
```

Comment:
why are we setting as 4:4:4 ? 

shouldn't this be dynamic based on data from v_frame

Reply:
Yes and no. 

The chroma_sampling of frame and encoder need to fit. I want the js user to be in charge to configure this, so she needs to watch out to configure encoder and frame fitting to each other.

Currently we are setting the chroma_sampling for the Frame in [rav1e_js/src/web/canvas.rs#L148](https://github.com/xiph/rav1e/pull/2446/files#diff-b3a48bb5a0fb1ce309a7685edb801abeR148).
In the (near) future I want the user to be able to decide for a chroma_sampling (I444, I422, I420) when sending an Image, or Video.

Reply:
Please document the same